ERCS/erc-7894.md

```diff
@@ -0,0 +1,143 @@
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+All functions and structures declared in this section MUST be used.
+
+**Non-Transferability Enforcement**
+   - **Disabled Functions**: `transferFrom` and `safeTransferFrom`
+   - **Description**: SocialSBT tokens are permanently non-transferable. The standard [ERC-721](./eip-721) transfer functions **MUST** be overridden and disabled to enforce soulbound behavior.
+
+1. **Minting Tokens**
+   - **Function**: mint() public override payable
+   - **Description**: The mint function MUST override the base [ERC-721](./eip-721) implementation to ensure compliance with SocialSBT’s non-transferability and single-token ownership per address.
+   - **Constraints**:
+     - The caller must pay the exact _price.
+     - Each address can own only one token.
+
+2. **Deleting Tokens**
+   - **Function**: deleteToken(uint256 tokenId) public
+   - **Description**: Allows token owners to permanently delete their tokens without refunds.
+   - **Constraints**:
+     - Caller must own the specified token.
+
+3. **Voting on Proposals**
+   - **Function**: vote(uint256 votingIndex_, bool choice_) public
+   - **Description**: Token holders can vote "yes" (true) or "no" (false) on proposals to adjust social points.
+   - **Constraints**:
+     - Caller must own a token.
+     - Each caller can vote only once per proposal.
+
+4. **Ending Voting**
+   - **Function**: endVoting(uint256 votingIndex_) public
+   - **Description**: Ends voting and adjusts points if the majority votes "yes."
+   - **Constraints**:
+     - Voting period must have expired.
+     - Proposal must be active.
+
+5. **Querying Token Points**
+   - **Function**: pointOf(uint256 tokenId) public view returns (uint256)
+   - **Description**: Retrieves the social points associated with a specified token.
+
+6. **Event Triggers**
+   - **NewVotingCreated**: Triggered when a voting proposal is created.
+   - **VoteEvent**: Triggered when a user casts a vote.
+   - **VotingEnd**: Triggered when voting ends.
+   - **PointUpdated**: Triggered when token points are adjusted.
+
+7. **Voting Struct**
+   - **votingIndex**: uint256
+   - **name**: string
+   - **description**: string
+   - **tokenIndex**: uint256
+   - **point**: uint256
+   - **increase**: bool
+   - **yes**: uint256
+   - **no**: uint256
+   - **startDate**: uint256
+   - **endDate**: uint256
+   - **isActive**: bool
```

Comment:
I'd recommend using a solidity interface instead of writing it out in this form.